src/thor/dijkstras.cc

```diff
@@ -581,32 +589,29 @@ void Dijkstras::ExpandForwardMultiModal(GraphReader& graphreader,
         mode_ = travel_mode_t::kPedestrian;
         walking_distance = 0;
         mode_change = true;
+      } else {
+        walking_distance += directededge->length();
+
+        // Prevent going from one transit connection directly to another
+        // at a transit stop - this is like entering a station and exiting
+        // without getting on transit
+        if (nodeinfo->type() == NodeType::kTransitEgress && pred.use() == Use::kTransitConnection &&
+            directededge->use() == Use::kTransitConnection) {
+          continue;
+        }
       }
 
       // Regular edge - use the appropriate costing and check if access
       // is allowed. If mode is pedestrian this will validate walking
       // distance has not been exceeded.
-      if (!mode_costing[static_cast<uint32_t>(mode_)]->Allowed(directededge, false, pred, tile,
-                                                               edgeid, 0, 0, restriction_idx)) {
+      if (!pc->Allowed(directededge, false, pred, tile, edgeid, 0, 0, restriction_idx) ||
+          walking_distance > max_walking_dist_) {
         continue;
       }
```

Comment:
the only thing that really changed here is that we're checking against the `max_walking_dist_` here. it didn't do that before and I'm not 100% we really want to do that for isochrones. IMO we should, but it's open for discussion.